packages/atomic/cypress/integration-insight-panel/route-mocks.ts

```diff
@@ -0,0 +1,157 @@
+export const insightSearchAlias = '@CoveoInsight';
+const searchRouteMatcher =
+  '**/rest/organizations/*/insight/v1/configs/*/search';
+
+export const interceptInsightSearch = (alias = insightSearchAlias) =>
+  cy.intercept(searchRouteMatcher).as(alias.substring(1));
+
+const inactiveLink = 'javascript:void(0);';
+const exampleSmartSnippetQuestion = 'Example smart snippet question';
+const exampleSmartSnippetSourceUri = inactiveLink;
+const exampleSmartSnippetSourceTitle = 'Example result title';
+const examplePermanentId = '123';
+const exampleInlineLinkText = 'Example inline link';
+const exampleInlineLinkUrl = inactiveLink;
+const exampleSmartSnippetAnswer = `
+    <div>
+      <p>Example smart snippet answer</p>
+      <a data-cy="smart-snippet__inline-link" href="${exampleInlineLinkUrl}">${exampleInlineLinkText}</a>
+    </div>
+  `;
+const exampleRelatedQuestions = [
+  {
+    question: 'first example question',
+    answerSnippet: exampleSmartSnippetAnswer,
+    title: 'first example title',
+    uri: inactiveLink,
+    documentId: {
+      contentIdKey: 'permanentid',
+      contentIdValue: '123',
+    },
+  },
+  {
+    question: 'second example question',
+    answerSnippet: exampleSmartSnippetAnswer,
+    title: 'second example title',
+    uri: inactiveLink,
+    documentId: {
+      contentIdKey: 'permanentid',
+      contentIdValue: '456',
+    },
+  },
+];
+
+export const mockSearchWithSmartSnippet = (
+  smartSnippetOptions: {
```

Comment:
Maybe you could piggy back on existing types definition from Headless for this whole file.

Something like: 

```ts
import type {SmartSnippetState} from '@coveo/headless/insight';
type MockOptions = Pick<SmartSnippetState, 'answer' | 'question' | 'etc....'>
```

Not sure how much simpler it would end up looking though

Reply:
Yeah I like the idea but it indeed might not be simpler. For this function alone I would need to make a union of 3 types, two of them using Pick. 😕